src/client.rs

```diff
@@ -200,15 +206,17 @@ impl Client {
     /// assert_eq!(result.len(), 1);
     /// assert_eq!(result["foo"], "42");
     /// ```
-    pub fn gets<V: FromMemcacheValueExt>(&self, keys: &[&str]) -> Result<HashMap<String, V>, MemcacheError> {
-        let mut con_keys: HashMap<usize, Vec<&str>> = HashMap::new();
+    pub fn gets<V, K, I>(&self, keys: I) -> Result<HashMap<String, V>, MemcacheError>
+    where
+        V: FromMemcacheValueExt,
+        K: AsRef<str>,
+        I: IntoIterator<Item = K>,
```

Comment:
I think by making the keys `IntoIterator`, we are adding unnecessary restrictions on ourselves (can't loop through keys multiple times) leading to allocations (ex: see [here](https://github.com/aisk/rust-memcache/pull/110/files#diff-253262251179a4073ac3610c6de21c77R359)) etc.

@aisk What do you think?

Reply:
I like ``IntoIterator`` because it lets a single method accept many kinds of key/value lists, such as ``&[]``, ``Vec``, ``HashMap``, or ``Option``. One alternative is to create our own ``KeyValueIter`` trait (that let us iterate over keys more than once, but over the values only once) and implement it for the collection types we care about. Would that be worth the effort?